bottlecap

```diff
@@ -0,0 +1,80 @@
+#!/bin/sh
+
+# bottlecap, 'cause it's kinda like cork... get it?
+
+usage() {
+	echo "usage:"
+	echo "bottlecap [--dev] [--runtime $runtime] [--build-dir $build_dir]  [--container $container] assemblerargs..."
+	echo "  --dev: rebuild and reinstall mantle, the build scripts, and the ostree-releng-scipts before running coreos-assembler"
+	echo "  --runtime: (auto) specify whether to use podman or docker. Defaults to podman if it is in \$PATH"
+	echo "  --build-dir: Where to put build artifacts, defaults to the current directory"
+	echo "  --container: which coreos-assembler container to use"
+	echo "  --help: print this help message"
+	echo "Command to be executed:"
+	runtime="echo $runtime"
+}
+
+dev=0
+help=0
+runtime="podman"
+which podman &> /dev/null || runtime="docker"
+build_dir="$(pwd)"
+container="quay.io/cgwalters/coreos-assembler"
+
+TEMP=$(getopt -o 'dr:b:c:h' --long 'dev,runtime:,build-dir:,container:,help' -- $@)
+eval set -- "$TEMP"
+
+while :
+do
+	case "$1" in
+		"-d"|"--dev")
+	  		dev=1
+			shift
+			;;
+		"-r"|"--runtime")
+			shift
+			runtime="$1"
+			shift
+			;;
+		"-b"|"--build-dir")
+			shift
+			build_dir="$(readlink -f $1)"
+			shift
+			;;
+		"-c"|"--container")
+			shift
+			container="$1"
+			shift
+			;;
+		"-h"|"--help")
+			help=1
+			shift
+			;;
+		--)
+			shift
+			break
+			;;
+		*)
+			echo "Error parsing args"
+			exit 1
+			;;
+	esac
+done
+
+if [ "$help" == 1 ]; then
+	usage
+fi
+
+script_dir="$(dirname $(readlink -f $0))"
+
+volumes="-v $build_dir:/srv "
+
+entrypoint=""
+
+if [ "$dev" == 1 ]; then
+	volumes+="-v $script_dir:/host/src "
+	volumes+="-v $(go env GOCACHE):/root/.cache/go-build "
+	entrypoint="--entrypoint /host/src/bottlecap-shim"
+fi
+
+$runtime run --rm --net=host -ti --privileged --userns=host $volumes --workdir /srv $entrypoint $container $@
```

Comment:
maybe we should `set -x` right above this command

also, do we expect users to run `sudo ./bottlecap` ?

Reply:
> also, do we expect users to run sudo ./bottlecap

If needed. I'd rather leave that to the user to decide than than bake it into the script.

Reply:
> maybe we should set -x right above this command

SGTM